src/status_im/extensions/ethereum.cljs

```diff
@@ -234,19 +234,51 @@
                         :on-failure on-failure}]
      (wrap-with-resolution db json-rpc-args :to execute-send-transaction))))
 
-(defn- parse-log [{:keys [address transactionHash blockHash transactionIndex topics blockNumber logIndex removed data]}]
-  (merge {:data    data
+(defn- event-topic-enc [event params]
+  (let [eventid (str event "(" (string/join "," params) ")")]
+    (abi-spec/sha3 eventid)))
+
+; Return a vector without elements of Y
+(defn diff-vectors [X Y]
+  (reduce #(remove (fn [x] (= x %2)) %1) X Y))
+
+; Return a map with all data params decoded
+(defn- decode-data [data events-hints first-topic indexed]
+  (let [index-event (.indexOf (mapv #(event-topic-enc (% :event) (% :params)) events-hints) first-topic)
+        event-hint (get events-hints index-event)
+        params-hint (diff-vectors (event-hint :params) indexed) ; Exclude indexed params from decode in data, these are decoded in topics
+        names-hint (mapv keyword (event-hint :names))
+        data-values (mapv (partial apply str) (partition 64 (string/replace-first data #"0x" "")))]
+    (zipmap names-hint (mapv abi-spec/hex-to-value data-values params-hint))))
+
+; This assumes that Topic 0 is filtered and return a map with all topics decoded
+(defn- decode-topics [topics indexed]
+  (zipmap [:topic1 :topic2 :topic3] (mapv abi-spec/hex-to-value topics indexed))) ; Solidity indexed event params may number up to 3 (max 4 Topics)
+
+; Return the vector of indexed types from the specified event (Topic0) from a vector of event hints given by the user
+(defn- get-indexed [hints first-topic]
+  ((get hints (.indexOf (mapv #(event-topic-enc (% :event) (% :params)) hints) first-topic)) :indexed))
```

Comment:
seems like same code is used on line 247
`(.indexOf (mapv #(event-topic-enc (% :event) (% :params)) events-hints) first-topic)`
It's also somewhat hard to read long lines with nested fn invokations, i suggest to consider using multi lines formatting or theading macros `-> ->> as->` whereever applicable.

```clj
((get hints (.indexOf (mapv #(event-topic-enc (% :event) (% :params)) hints) first-topic)) :indexed)
```
becomes
```clj
(as-> hints $
    (mapv #(event-topic-enc (% :event) (% :params)) $)
    (.indexOf $ first-topic)
    (get $ hints)
    (get $ :indexed))
```

Reply:
Hi @dymnz I updated it with threading macros, thanks for your suggestion :+1: